src/test/ui/pattern/usefulness/issue-78123-non-exhaustive-reference.rs

```diff
@@ -0,0 +1,7 @@
+enum A {}
+
+fn f(a: &A) {
+    match a {} //~ ERROR non-exhaustive patterns: type `&A` is non-empty
```

Comment:
You can also add `//~ NOTE` comments for notes. I think it's important here since that's what we're testing.
The way it works is that if there are no `//~ NOTE ...` comments then compiletest ignores notes, but if there is one the it will check we got them all, like it does for warnings.